Create release notes for 1.17 Xinzhou (#631)
kyma-project/website: content/blog-posts/2020-11-19-release-notes-1.17/index.md

## Website

### New Getting Started guides

Have you ever wondered how to start with Kyma? How to connect all the dots (components) into a meaningful scenario? We tried to address such questions with a set of [Getting Started guides](https://kyma-project.io/docs/1.17/root/getting-started/) we are happy to share with you. These guides lead you through an end-to-end scenario that shows a sample Kyma use case. After completing it, you will get a Function and a microservice that react to events sent from an external application and use the external Redis service to store the events data. These guides show all Kyma’s major features, introduce you to our components, and give you a chance to get to know Kyma either through the kubectl commands or the Console UI steps.

## CLI

### Upgrade command

We introduced the new command as a first step to offer a smooth Kyma upgrade. To use it, download the CLI in the same version as the target Kyma version and run [`kyma upgrade`](/cli/commands/#kyma-upgrade-kyma-upgrade). The command will perform all the steps for you. Additionally, it offers a way to customize the component and overrides.

### Command to create a system

> Another new command that we added is [`kyma create system`](/cli/commands/#kyma-create-system-kyma-create-system). In just one step, it allows you to create a system (an Application), bind it to a Namespace, and get a token to pair the system with an external application.

Reviewer: Our guidelines say:
```
"Application" to describe an external solution connected to Kyma through the Application Connector, "application" to describe software
```
So I guess it would be the other way round?

Reviewer 2: It's written in lowercase in the [CLI docs](https://kyma-project.io/docs/master/cli/commands/#kyma-create-system-kyma-create-system-detailed-description), so I'd say we go with lowercase... however, I am really confused with the differentiation here. And if we are confused, so are the contributors/users. I've added this topic to the next SIG Content agenda so we can discuss it :)

### Manage functions

We released a set of new commands for function developers. These commands are very useful when starting a function from scratch or whenever you want to apply code or configuration changes to a function.

Find more details in the dedicated [tutorial](https://kyma-project.io/docs/master/cli/overview/#tutorials-use-kyma-cli-to-manage-functions).

### PR number support for install command

> The `install` command now supports installing Kyma based on a specific pull request (PR) to the Kyma repository. So `kyma install --source=PR-XXXX` will install Kyma to your Kubernetes cluster using the revision identified by the latest commit to the specific PR.

Reviewer: ```suggestion
The `install` command now supports installing Kyma based on a specific pull request (PR) submitted to the Kyma repository. So `kyma install --source=PR-XXXX` will install Kyma on your Kubernetes cluster using the revision identified by the latest commit to the specific PR.
```

### Minikube support when using a DNS-Proxy on MacOS

We added a new flag to the Minikube command to support a setup which is typical for corporate environments. The new flag is: `kyma provision minikube --use-hyperkit-vpnkit-sock`.

## Monitoring

### AuthProxy support for Grafana

The Grafana chart now offers an option to deploy Grafana along with an instance of a Keycloak Gatekeeper (`louketo-proxy`) in front of it. Enabling the proxy and switching to Grafana in AuthProxy allows you to control the access by the `group` claim of your IDToken.

> ### Removed dashboards for CoreDNS/KubeDNS and Kube-Proxy

Reviewer: ```suggestion
### Removed dashboards for CoreDNS, KubeDNS, and Kube-Proxy
```

> The Grafana bundle in Kyma brings a lot of ready-to-use dashboards together with service monitors definitions for the related components. Here, the focus is on having an observable Kyma ecosystem. Among those dashboards are CoreDNS/KubeDNS and Kube-Proxy. The related components are provided by Kubernetes on which Kyma is installed, so they are not shipped with Kyma. Different Kubernetes providers are changing settings in the setup of the components, and as we do not offer support for the mentioned components, they will not be activated by default. You can still easily enable them via the configuration options of the Grafana sub-chart, but that might require further configuration dependent on your Kubernetes provider.

Reviewer: I would change this part somehow as it is not that clear: `The related components are provided by Kubernetes on which Kyma is installed`.

> The Grafana bundle in Kyma brings a lot of ready-to-use dashboards together with service monitors definitions for the related components. Here, the focus is on having an observable Kyma ecosystem. Among those dashboards are CoreDNS/KubeDNS and Kube-Proxy. The related components are provided by Kubernetes on which Kyma is installed, so they are not shipped with Kyma. Different Kubernetes providers are changing settings in the setup of the components, and as we do not offer support for the mentioned components, they will not be activated by default. You can still easily enable them via the configuration options of the Grafana sub-chart, but that might require further configuration dependent on your Kubernetes provider.

Author: Thanks, I changed it in https://github.com/kyma-project/website/pull/631/commits/90ac644774cd106db815b270766d21c023f4c489

## Tracing

### Option to disable sending traces

The new `global.tracing.enabled` override is available in the Event Sources Controller Manager sub-chart configuration. By setting it to `false`, you can disable sending traces from Kyma components to the service deployed by the Tracing component.

### Improved integration of Loki to Grafana

If log lines are in a JSON format, the root elements of the document will now be treated as their own log record attributes in Fluent Bit. With that, the root elements will be shown as fields in Grafana.

> To be consistent with the Prometheus labeling, the `instance` label got renamed to `pod`, as it contains the Pod name.

Reviewer: ```suggestion
To be consistent with the Prometheus labeling, the `instance` label got renamed to `pod` as it contains the Pod name.
```

## Kiali

### General update and Jaeger integration

> Kiali got upgraded to the latest version 1.24. With that, proper configuration of the component status was possible and there is no red light shown in the top bar. Furthermore, the metric dashboards work more reliably.

Reviewer: ```suggestion
Kiali got upgraded to the latest 1.24 version. With that, proper configuration of the component status was possible and there is no red light shown in the top bar. Furthermore, the metric dashboards operate more smoothly.
```

> Additionally, we now configured Jaeger so that you can see traces inline in Kiali with deep-linking into the Jaeger UI.

Reviewer: ```suggestion
Additionally, we now configured Jaeger so that you can see inline traces in Kiali with deep-linking into the Jaeger UI.
```

## Eventing

> ### Removed Knative-Serving chart

Reviewer: ```suggestion
### Removed the Knative-Serving chart
```

> In release 1.17, we remove the support for Knative-Serving from Kyma. After the upgrade, Knative-Serving will still be fully functional in the cluster, but it can be removed easily in order to free resources. Removing Knative-Serving will not affect supported Kyma functionality. To learn more, read the [migration guide](https://github.com/kyma-project/kyma/blob/release-1.17/docs/migration-guides/1.15-1.17.md).

Reviewer: ```suggestion
In release 1.17, we remove the support for Knative-Serving from Kyma. After the upgrade, Knative-Serving will still be fully functional in the cluster, but it can be removed easily in order to free resources. Removing Knative-Serving will not affect the supported Kyma functionality. To learn more, read the [Migration Guide](https://github.com/kyma-project/kyma/blob/release-1.17/docs/migration-guides/1.15-1.17.md).
```

## Installation

### Istio upgrade to 1.7.4

> With this release, we have upgraded Istio from 1.5.10 to 1.7.4. This version of Istio requires Kubernetes 1.16 or higher.  Find more details in the [migration guide](https://github.com/kyma-project/kyma/blob/master/docs/migration-guides/1.15-1.17.md) and the [Istio 1.7.4 release notes](https://istio.io/latest/news/releases/1.7.x/announcing-1.7/).

Reviewer: ```suggestion
With this release, we have upgraded Istio from 1.5.10 to 1.7.4. This version of Istio requires Kubernetes 1.16 or higher. Find more details in the [Migration Guide](https://github.com/kyma-project/kyma/blob/master/docs/migration-guides/1.15-1.17.md) and the [Istio 1.7.4 release notes](https://istio.io/latest/news/releases/1.7.x/announcing-1.7/).
```

Two known issues exist as a result of this upgrade:

> 

Reviewer: Don't we want to place known issues somewhere in a separate section as in the [template](https://github.com/kyma-project/community/blob/master/guidelines/templates/resources/release-notes.md#known-issues)?

Reviewer: I can see that you extracted these two issues to separate sections as per the template so these two below should be removed - maybe it would be just enough to add some not that the upgrade causes some known issues and link to the new sections?

> * Prometheus-Istio crashes:

Reviewer: ```suggestion
* Prometheus-Istio crashes
```

With the switch to Istio telemetry v2 (required for Istio 1.7), an important feature about metric retention is still missing in the Istio-proxy. Due to that, under heavy service topology changes there might be proxy instances such as ORY Oathkeeper that keep orphaned metrics scraped by the prometheus-istio instance. That can cause a OOM crash. Click [here](https://github.com/kyma-project/kyma/issues/9867) to learn more.

> * Kiali Graph and App Metrics missing:

Reviewer: ```suggestion
* Kiali Graph and App Metrics missing
```

In order to mitigate the Prometheus-Istio crashes problem, labels for Istio metrics were reduced to a bare minimum. With that, Kiali is no longer showing the full graph. Click [here](https://github.com/kyma-project/kyma/issues/9886) for information on how to enable all labels on the istio metrics in order to see the graph again.

## Console

### New features on namespace details page

> We have added new functionalities to the namespace overview page. They allow you to easily jump from the namespace overview to the Grafana dashboard and inspect logs from the namespace. You can also see all resource quotas and limit ranges that apply to the namespace:

Reviewer: ```suggestion
We have added new functionalities to the Namespace overview page. They allow you to easily jump from the Namespace overview to the Grafana dashboard and inspect logs from that Namespace. You can also see all resource quotas and limit ranges that apply to the Namespace:
```

![Namespace overview](./namespace-overview.png)

> Additionally, you can edit those limits via a handy YAML editor that slides out as a drawer:

Reviewer: ```suggestion
Additionally, you can edit those limits using a handy YAML editor that slides out as a drawer:
```

## Website

### New Getting Started guides

Have you ever wondered how to start with Kyma? How to connect all the dots (components) into a meaningful scenario? We tried to address such questions with a set of [Getting Started guides](https://kyma-project.io/docs/1.17/root/getting-started/) we are happy to share with you. These guides lead you through an end-to-end scenario that shows a sample Kyma use case. After completing it, you will get a Function and a microservice that react to events sent from an external application and use the external Redis service to store the events data. These guides show all Kyma’s major features, introduce you to our components, and give you a chance to get to know Kyma either through the kubectl commands or the Console UI steps.

## CLI

### Upgrade command

We introduced the new command as a first step to offer a smooth Kyma upgrade. To use it, download the CLI in the same version as the target Kyma version and run [`kyma upgrade`](/cli/commands/#kyma-upgrade-kyma-upgrade). The command will perform all the steps for you. Additionally, it offers a way to customize the component and overrides.

### Command to create a system

> Another new command that we added is [`kyma create system`](/cli/commands/#kyma-create-system-kyma-create-system). In just one step, it allows you to create a system (an application), bind it to a Namespace, and get a token to pair the system with an external application.

Reviewer: ```suggestion
Another new command that we added is [`kyma create system`](/cli/commands/#kyma-create-system-kyma-create-system). In just one step, it allows you to create a system (an Application), bind it to a Namespace, and get a token to pair the system with an external application.
```

To be consistent with the Prometheus labeling, the `instance` label got renamed to `pod` as it contains the Pod name.

## Kiali

### General update and Jaeger integration

Kiali got upgraded to the latest 1.24 version. With that, proper configuration of the component status was possible and there is no red light shown in the top bar. Furthermore, the metric dashboards operate more smoothly.
Additionally, we now configured Jaeger so that you can see inline traces in Kiali with deep-linking into the Jaeger UI.

## Eventing

### Removed the Knative-Serving chart

In release 1.17, we removed the support for Knative-Serving from Kyma. After the upgrade, Knative-Serving will still be fully functional in the cluster, but it can be removed easily in order to free resources. Removing Knative-Serving will not affect the supported Kyma functionality. To learn more, read the [Migration Guide](https://github.com/kyma-project/kyma/blob/release-1.17/docs/migration-guides/1.15-1.17.md).

## Installation

### Istio upgrade to 1.7.4

> With this release, we have upgraded Istio from 1.5.10 to 1.7.4. This version of Istio requires Kubernetes 1.16 or higher. Find more details in the [Migration Guide](https://github.com/kyma-project/kyma/blob/release-1.17/docs/migration-guides/1.15-1.17.md) and the [Istio 1.7.4 release notes](https://istio.io/latest/news/releases/1.7.x/announcing-1.7/). This upgrade resulted in two known issues which are documented [here](#known-issues).

Reviewer: ```suggestion
With this release, we have upgraded Istio from 1.5.10 to 1.7.4. This version of Istio requires Kubernetes 1.16 or higher. Find more details in the [Migration Guide](https://github.com/kyma-project/kyma/blob/release-1.17/docs/migration-guides/1.15-1.17.md) and the [Istio 1.7.4 release notes](https://istio.io/latest/news/releases/1.7.x/announcing-1.7/). This upgrade resulted in two [known issues](#known-issues).
```

## CLI

### Upgrade command

We introduced the new command as a first step to offer a smooth Kyma upgrade. To use it, download the CLI in the same version as the target Kyma version and run [`kyma upgrade`](/cli/commands/#kyma-upgrade-kyma-upgrade). The command will perform all the steps for you. Additionally, it offers a way to customize the component and overrides.

### Command to create a system

Another new command that we added is [`kyma create system`](/cli/commands/#kyma-create-system-kyma-create-system). In just one step, it allows you to create a system (an Application), bind it to a Namespace, and get a token to pair the system with an external application.

### Manage Functions

We released a set of new commands for Function developers. These commands are very useful when creating a Function from scratch or whenever you want to apply code or configuration changes to a Function.

> Find more details in the dedicated [tutorial](https://kyma-project.io/docs/master/cli/overview/#tutorials-use-kyma-cli-to-manage-functions).

Reviewer: It should not point to the master but to the 1.17 branch (similarly as all other links)

> Find more details in the dedicated [tutorial](https://kyma-project.io/docs/master/cli/overview/#tutorials-use-kyma-cli-to-manage-functions).

Reviewer: ```suggestion
Find more details in the dedicated [tutorial](https://kyma-project.io/docs/1.17/cli/overview/#tutorials-use-kyma-cli-to-manage-functions).
```

## Tracing

### Option to disable sending traces

The new `global.tracing.enabled` override is available in the Event Sources Controller Manager sub-chart configuration. By setting it to `false`, you can disable sending traces from Kyma components to the service deployed by the Tracing component.

### Improved integration of Loki to Grafana

If log lines are in a JSON format, the root elements of the document will now be treated as their own log record attributes in Fluent Bit. With that, the root elements will be shown as fields in Grafana.

To be consistent with the Prometheus labeling, the `instance` label got renamed to `pod` as it contains the Pod name.

## Kiali

### General update and Jaeger integration

Kiali got upgraded to the latest 1.24 version. With that, proper configuration of the component status was possible and there is no red light shown in the top bar. Furthermore, the metric dashboards operate more smoothly.
Additionally, we now configured Jaeger so that you can see inline traces in Kiali with deep-linking into the Jaeger UI.

## Eventing

### Removed the Knative-Serving chart

> In release 1.17, we remove the support for Knative-Serving from Kyma. After the upgrade, Knative-Serving will still be fully functional in the cluster, but it can be removed easily in order to free resources. Removing Knative-Serving will not affect the supported Kyma functionality. To learn more, read the [Migration Guide](https://github.com/kyma-project/kyma/blob/release-1.17/docs/migration-guides/1.15-1.17.md).

Reviewer: ```suggestion
In release 1.17, we removed the support for Knative-Serving from Kyma. After the upgrade, Knative-Serving will still be fully functional in the cluster, but it can be removed easily in order to free resources. Removing Knative-Serving will not affect the supported Kyma functionality. To learn more, read the [Migration Guide](https://github.com/kyma-project/kyma/blob/release-1.17/docs/migration-guides/1.15-1.17.md).
```

## Installation

### Istio upgrade to 1.7.4

> With this release, we have upgraded Istio from 1.5.10 to 1.7.4. This version of Istio requires Kubernetes 1.16 or higher. Find more details in the [Migration Guide](https://github.com/kyma-project/kyma/blob/master/docs/migration-guides/1.15-1.17.md) and the [Istio 1.7.4 release notes](https://istio.io/latest/news/releases/1.7.x/announcing-1.7/).

Reviewer: ```suggestion
With this release, we have upgraded Istio from 1.5.10 to 1.7.4. This version of Istio requires Kubernetes 1.16 or higher. Find more details in the [Migration Guide](https://github.com/kyma-project/kyma/blob/release-1.17/docs/migration-guides/1.15-1.17.md) and the [Istio 1.7.4 release notes](https://istio.io/latest/news/releases/1.7.x/announcing-1.7/).
```

Two known issues exist as a result of this upgrade:

* Prometheus-Istio crashes
With the switch to Istio telemetry v2 (required for Istio 1.7), an important feature about metric retention is still missing in the Istio-proxy. Due to that, under heavy service topology changes, there might be proxy instances such as ORY Oathkeeper that keep orphaned metrics scraped by the prometheus-istio instance. That can cause an OOM crash. See [issue 9867](https://github.com/kyma-project/kyma/issues/9867) to learn more.

* Kiali Graph and App Metrics missing
In order to mitigate the Prometheus-Istio crashes problem, labels for Istio metrics were reduced to a bare minimum. With that, Kiali is no longer showing the full graph. See [issue 9886](https://github.com/kyma-project/kyma/issues/9886) for information on how to enable all labels on the Istio metrics in order to see the graph again.

## Console

### New features on the Namespace details page

> We have added new functionalities to the Namespace overview page. They allow you to easily jump from the Namespace overview to the Grafana dashboard and inspect logs from that Namespace. You can also see all resource quotas and limit ranges that apply to the Namespace:

Reviewer: ```suggestion
We have added new features to the Namespace overview page. They allow you to easily jump from the Namespace overview to the Grafana dashboard and inspect logs from that Namespace. You can also see all resource quotas and limit ranges that apply to the Namespace:
```
